pkg/resource/deploy/snapshot.go

```diff
@@ -155,6 +155,16 @@ func (snap *Snapshot) VerifyIntegrity() error {
 					}
 					return fmt.Errorf("child resource %s refers to missing parent %s", urn, par)
 				}
+
+				// Ensure that our URN is a child of the parent's URN.
```

Comment:
We probably want to comment this block out to start with, just in case anyone has an invalid state (according to this) that they want to still try and use, but after a couple of releases we could probably start checking this. If anyone has issues with that we can just tell them to update using one release before.

Reply:
Could we warn it's changing and continue, then in a few releases move to error? Or possibly just convert them. I have seen customers do like 10 minor version jumps

Reply:
It's hard to do the conversion at this point. We could warn for now, but at some point I think this just needs to be a validation error for our sanity. Most stacks _shouldn't_ be in a state that they hit this anyway, but if people do have half deployed things they would see this.

Reply:
That is good to know on conversion. I definitely agree with erroring out. I was just thinking warnings so we could start getting messages out now something is wrong in the state.